kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h

```diff
@@ -1177,6 +1178,77 @@ class ResidualBasedBlockBuilderAndSolver
         KRATOS_CATCH("")
     }
 
+    /**
+     * @brief It reconstructs the solution vector at slave DOFs
+     * @param pScheme The pointer to the integration scheme
+     * @param rModelPart The model part to compute
+     * @param rA The LHS matrix of the system of equations
+     * @param rDx The vector of unkowns
+     * @param rb The RHS vector of the system of equations
+     */
+    void ReconstructSlaveSolution(
+        typename TSchemeType::Pointer pScheme,
+        ModelPart& rModelPart,
+        TSystemMatrixType& rA,
+        TSystemVectorType& rDx,
+        TSystemVectorType& rb
+        ) override
+    {
+        KRATOS_TRY
+
+        auto& r_master_slave_constraints = rModelPart.MasterSlaveConstraints();
+        // Reset slave dofs
+        block_for_each(r_master_slave_constraints, [&rDx](const MasterSlaveConstraint &r_master_slave_constraint) {
+            const auto &r_slave_dofs_vector = r_master_slave_constraint.GetSlaveDofsVector();
+            for (const auto &r_slave_dof : r_slave_dofs_vector)
+            {
+                #pragma omp atomic
+                rDx[r_slave_dof->EquationId()] *= 0.0;
+            }
+        });
+
+        // Apply constraints
+        struct TLS
+        {
+            Matrix relation_matrix;
+            Vector constant_vector;
+            Vector master_dofs_values;
+        };
+
+        block_for_each(r_master_slave_constraints, TLS(), [&rDx, &rModelPart](const MasterSlaveConstraint &r_master_slave_constraint, TLS &rTLS) {
+            // Detect if the constraint is active or not. If the user did not make any choice the constraint
+            // It is active by default
+            bool constraint_is_active = true;

```

Comment:
`const bool constraint_is_active = r_master_slave_constraint.IsDefined(ACTIVE) ? r_master_slave_constraint.Is(ACTIVE) : true ;
`

Reply:
:+1

Reply:
:+1: